packages/logger/src/logger/PersistentTransport.ts

```diff
@@ -59,8 +61,15 @@ export abstract class PersistentTransport extends Transport {
   }
 
   // Signal to pause processing messages from persistent storage.
-  pauseProcessing() {
+  async pauseProcessing() {
```

Comment:
Could there be a race condition in `pauseProcessing` due to tasks starting between the check of `isQueueBeingExecuted` and setting `canProcess` to `false`, potentially processing logs post-pause?

Reply:
I don't think that should be possible as all sync operations within async function are still executed within a single thread without context switching. So nothing should interrupt between setting `canProcess` to `false` and the check of `isQueueBeingExecuted`